pkg/visor/visor.go

```diff
@@ -302,6 +306,63 @@ func (node *Node) Start() error {
 	return nil
 }
 
+// RunDaemon starts a skywire-peering-daemon as an external process
+func (node *Node) RunDaemon() error {
+	node.Onspd = true
+	bin, err := exec.LookPath("daemon")
+	if err != nil {
+		return fmt.Errorf("Cannot find `skywire-peering-daemon` binary in $PATH: %s", err)
+	}
+
+	dir, err := ioutil.TempDir("", "named_pipes")
+	if err != nil {
+		return fmt.Errorf("Couldn't create named_pipes dir: %s", err)
+	}
+
+	namedPipe := filepath.Join(dir, "stdout")
+	lAddr := node.conf.STCP.LocalAddr
+	pubKey := node.conf.Node.StaticPubKey.Hex()
+	err = syscall.Mkfifo(namedPipe, 0600)
+	if err != nil {
+		return err
+	}
+
+	node.spdCmd = exec.Command(bin, pubKey, lAddr, namedPipe)
+	if err := execute(node.spdCmd); err != nil {
+		return fmt.Errorf("Failed to start daemon as an external process: %s", err)
+	}
+
+	node.logger.Info("Opening named pipe for reading packets from skywire-peering-daemon")
+	stdOut, err := os.OpenFile(namedPipe, os.O_RDONLY, 0600)
+	if err != nil {
+		return err
+	}
+
+	pubKeyTable := node.conf.STCP.PubKeyTable
```

Comment:
We may need to check whether the pk table is nil or not.

Reply:
You're right.